versioned_hdf5/tools.py

```diff
@@ -33,7 +33,9 @@ def asarray(a, dtype=None):
         # np.array(-1).astype("u1") doesn't raise and returns 255!
         return a.view(dtype)
 
-    return a.astype(dtype)
+    if hasattr(a, "astype"):
+        return a.astype(dtype)
+    return np.asarray(a, dtype=dtype)
```

Comment:
I could have also added
```python
if hasattr(a, "__array_function__`):
    return np.asarray(a, dtype=dtype, like=a)
```
but it wasn't terribly interesting as a use case in real life.